tests/test-tokenizer-0.cpp

```diff
@@ -5,16 +5,31 @@
 #include <map>
 #include <vector>
 
+static std::string unescape_whitespace(llama_context* ctx, const llama_token* tokens, int count) {
+    std::string result;
+    for (int i = 0; i < count; ++i) {
+        result += llama_token_to_str(ctx, tokens[i]);
+    }
+    return result;
+}
+
 static const std::map<std::string, std::vector<llama_token>> & k_tests()
 {
     static std::map<std::string, std::vector<llama_token>> _k_tests = {
-        { "Hello World",        { 1,  10994,   2787, }, },
-        { " Hello World",       { 1,  15043,   2787, }, },
-        { " Hello World!",      { 1,  15043,   2787,  29991, }, },
-        { " this is 🦙.cpp",    { 1,    445,    338,  29871,    243,    162,    169,    156,  29889,   8223, }, },
-        { "w048 7tuijk dsdfhu", { 1,  29893,  29900,  29946,  29947,  29871,  29955,   9161,  13535,  18031,   2176,   6905, }, },
-        { "нещо на Български",  { 1,    821,   4851,    665,   1386,  29713,   1305, }, },
-    };
+        { " ",                  {1,    259, }, },
+        { "\t",                 { 1,    29871,   12, }, },
+        { "\n",                 { 1,    29871,   13, }, },
+        { "\t\n",               { 1,    29871,   12,     13, }, },
+        { "Hello world",        { 1,  15043,   3186, }, },
+        { " Hello world",       { 1,  29871,  15043,   3186, }, },
+        { "Hello World",        { 1,  15043,   2787, }, },
+        { " Hello World",       { 1,  29871,  15043,   2787, }, },
+        {" Hello World!",       { 1,  29871,  15043,   2787,  29991, }, },
+        {" this is 🦙.cpp",    { 1,  29871,    445,    338,  29871,    243,    162,    169,    156,  29889,   8223, }, },
+        {"w048 7tuijk dsdfhu",  { 1,    281,  29900,  29946,  29947,  29871,  29955,   9161,  13535,  18031,   2176,   6905, }, },
+        {"нещо на Български",   { 1,   1538,   4851,    665,   1386,  29713,   1305, }, },
+        {"How are you?",        { 1,   1128,    526,    366,  29973, }, },
+     };
```

Comment:
I recommend adding two tests:
- with emojis 🚀 (normal) 😶‍🌫️ (multiple emojis concatenated) ✅ (only emoji that has its own token)
- “កាន់តែពិសេសអាចខលចេញ” this script that contains characters that are tokens and some that are not. As well as ligatures where ក next to 'ា will turn into កា.

Reply:
Just checked and I get the same results for the two.